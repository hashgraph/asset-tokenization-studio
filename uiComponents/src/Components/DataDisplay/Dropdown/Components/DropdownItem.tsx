import type { PhosphorIconProps } from "@/Components/Foundations/PhosphorIcon";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { Text } from "@/Components/Foundations/Text";
import type { MenuItemProps as ChakraMenuItemProps } from "@chakra-ui/react";
import { MenuItem as ChakraMenuItem } from "@chakra-ui/react";
import React from "react";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
import type { DropdownThemeStyle } from "../Dropdown";

export interface DropdownItemProps extends ChakraMenuItemProps {
  label: string;
  icon?: PhosphorIconProps["as"];
  isDisabled?: boolean;
  isActive?: boolean;
}
export const DropdownItem = ({
  label,
  icon,
  isDisabled,
  isActive,
  ...props
}: DropdownItemProps) => {
  const styles = useChakraMultiStyleConfig("Dropdown", {
    isActive,
    isDisabled,
  }) as DropdownThemeStyle;

  return (
    <ChakraMenuItem
      data-testid={"dropdown-item"}
      disabled={isDisabled}
      {...props}
      sx={styles.itemContainer}
    >
      {icon && (
        <PhosphorIcon
          data-testid="dropdown-item-icon"
          as={icon}
          size="xxs"
          weight={styles.itemIconWeight}
        />
      )}
      <Text sx={styles.itemLabel}>{label}</Text>
    </ChakraMenuItem>
  );
};
