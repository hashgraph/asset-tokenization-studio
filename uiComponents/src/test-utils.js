import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
