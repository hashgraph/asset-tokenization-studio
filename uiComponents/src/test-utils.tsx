import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
