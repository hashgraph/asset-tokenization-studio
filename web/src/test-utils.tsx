import React from "react";
import { ChakraProvider } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render, waitFor } from "@testing-library/react";
import type { MemoryHistory } from "history";
import { createMemoryHistory } from "history";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import i18n from "./i18n";
import theme from "./theme";
import userEvent from "@testing-library/user-event";

export const selectCalendar = async (
  component: RenderResult,
  id: string,
  day: string | number = new Date().getDate(),
) => {
  const calendar = component.getByTestId(id);
  await userEvent.click(calendar);

  await waitFor(() => {
    const daysToSelect = component.getAllByTestId(`day-${day}`);

    const dayToSelect = daysToSelect.find(
      (day) => !day.hasAttribute("disabled"),
    );

    if (dayToSelect) {
      userEvent.click(dayToSelect);
    }
  });

  await userEvent.click(document.body);
};

export const testQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnMount: true,
      cacheTime: 0,
    },
  },
  logger: {
    log: console.log,
    warn: console.warn,
    // âœ… no more errors on the console for tests
    error: () => {},
  },
});

export const queryWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

const memoryHistory = createMemoryHistory();

export const AllProviders = ({
  children,
  history = memoryHistory,
}: {
  children?: React.ReactNode;
  history?: MemoryHistory;
}) => {
  return (
    <I18nextProvider i18n={i18n}>
      <QueryClientProvider client={testQueryClient}>
        <ChakraProvider theme={theme}>
          <Router location={history.location} navigator={history}>
            {children}
          </Router>
        </ChakraProvider>
      </QueryClientProvider>
    </I18nextProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  {
    options,
    history,
  }: {
    options?: RenderOptions;
    history?: MemoryHistory;
  } = {},
): RenderResult =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllProviders history={history}>{children}</AllProviders>
    ),
    ...options,
  });

export { customRender as render };
