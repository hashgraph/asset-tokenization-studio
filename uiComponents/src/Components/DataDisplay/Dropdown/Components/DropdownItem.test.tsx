import { render } from "@/test-utils";
import { Cookie } from "@phosphor-icons/react";
import React from "react";
import type { DropdownItemProps } from "./DropdownItem";
import { DropdownItem } from "./DropdownItem";
import { fireEvent } from "@testing-library/react";
import { Menu } from "@chakra-ui/react";

describe(`<DropdownItem />`, () => {
  const defaultProps: DropdownItemProps = {
    label: "Item",
    icon: Cookie,
  };
  const factoryComponent = (props: DropdownItemProps = defaultProps) =>
    render(
      <Menu>
        <DropdownItem {...props} />
      </Menu>
    );

  test("renders correctly", () => {
    const component = factoryComponent();
    expect(component.getByTestId("dropdown-item-icon")).toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot();
  });

  test("renders correctly without icon", () => {
    const component = factoryComponent({ label: "Item" });
    expect(
      component.queryByTestId("dropdown-item-icon")
    ).not.toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot("WithoutIcon");
  });

  test("renders correctly when hovered", async () => {
    const component = factoryComponent();
    fireEvent.mouseEnter(component.getByTestId(`dropdown-item`));

    expect(component.asFragment()).toMatchSnapshot("Hovered");
  });

  test("renders correctly when it's disabled", async () => {
    const component = factoryComponent({ ...defaultProps, isDisabled: true });

    expect(component.asFragment()).toMatchSnapshot("Disabled");
  });

  test("renders correctly when it's active", async () => {
    const component = factoryComponent({ ...defaultProps, isActive: true });

    expect(component.asFragment()).toMatchSnapshot("Active");
  });
});
