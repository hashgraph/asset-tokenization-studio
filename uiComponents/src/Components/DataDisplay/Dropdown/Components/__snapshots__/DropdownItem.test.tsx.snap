// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem /> renders correctly 1`] = `
<DocumentFragment>
  <button
    class="chakra-menu__menuitem css-jtq5p1"
    data-index="0"
    data-testid="dropdown-item"
    id="menu-list-:r1:-menuitem-:r2:"
    role="menuitem"
    tabindex="-1"
    type="button"
  >
    <svg
      class="chakra-icon css-kpbtv0"
      data-testid="dropdown-item-icon"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
      />
    </svg>
    <p
      class="css-1a1j0go"
    >
      Item
    </p>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<DropdownItem /> renders correctly when hovered: Hovered 1`] = `
<DocumentFragment>
  <button
    class="chakra-menu__menuitem css-jtq5p1"
    data-index="0"
    data-testid="dropdown-item"
    id="menu-list-:r7:-menuitem-:r8:"
    role="menuitem"
    tabindex="0"
    type="button"
  >
    <svg
      class="chakra-icon css-kpbtv0"
      data-testid="dropdown-item-icon"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
      />
    </svg>
    <p
      class="css-1a1j0go"
    >
      Item
    </p>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<DropdownItem /> renders correctly when it's active: Active 1`] = `
<DocumentFragment>
  <button
    class="chakra-menu__menuitem css-mbp4z0"
    data-index="0"
    data-testid="dropdown-item"
    id="menu-list-:rd:-menuitem-:re:"
    role="menuitem"
    tabindex="-1"
    type="button"
  >
    <svg
      class="chakra-icon css-kpbtv0"
      data-testid="dropdown-item-icon"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224,120a40,40,0,0,1-40-40,8,8,0,0,0-8-8,40,40,0,0,1-40-40,8,8,0,0,0-8-8A104,104,0,1,0,232,128,8,8,0,0,0,224,120ZM75.51,99.51a12,12,0,1,1,0,17A12,12,0,0,1,75.51,99.51Zm25,73a12,12,0,1,1,0-17A12,12,0,0,1,100.49,172.49Zm23-40a12,12,0,1,1,17,0A12,12,0,0,1,123.51,132.49Zm41,48a12,12,0,1,1,0-17A12,12,0,0,1,164.49,180.49Z"
      />
    </svg>
    <p
      class="css-1a1j0go"
    >
      Item
    </p>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<DropdownItem /> renders correctly when it's disabled: Disabled 1`] = `
<DocumentFragment>
  <button
    class="chakra-menu__menuitem css-1j5s9kt"
    data-index="0"
    data-testid="dropdown-item"
    id="menu-list-:ra:-menuitem-:rb:"
    role="menuitem"
    tabindex="-1"
    type="button"
  >
    <svg
      class="chakra-icon css-kpbtv0"
      data-testid="dropdown-item-icon"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
      />
    </svg>
    <p
      class="css-80t0jy"
    >
      Item
    </p>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<DropdownItem /> renders correctly without icon: WithoutIcon 1`] = `
<DocumentFragment>
  <button
    class="chakra-menu__menuitem css-jtq5p1"
    data-index="0"
    data-testid="dropdown-item"
    id="menu-list-:r4:-menuitem-:r5:"
    role="menuitem"
    tabindex="-1"
    type="button"
  >
    <p
      class="css-1a1j0go"
    >
      Item
    </p>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
